### PR TITLE
AK+LibWeb+LibXML: XHTML Says Hi!

### PR DESCRIPTION
**Pull request by alimpfard**
LibXML status: Not very good at XML'ing XML things, but good enough at XML'ing XHTML things.

Needs a bit of work to cleanup, maybe some De-hack-ification, but one image:
image

As far as XML itself goes, it:
- Reads DTDs (but does not verify anything with them)
- Resolves entity references
- Does _not_ resolve PE references
- XML conformance test suite: `763 passed, 167 failed, 238 runner failed of 1168 tests run.`

### CODE REVIEW

File: Userland/Libraries/LibWeb/HTML/DOMParser.cpp [hunk 1]
```diff
@@ -36,16 +37,23 @@ NonnullRefPtr<DOM::Document> DOMParser::parse_from_string(String const& string,
...
-        //           2. Let root be the result of creating an element given document, "parsererror", and "http://www.mozilla.org/newlayout/xml/parsererror.xml".
-        //           3. Optionally, add attributes or children to root to describe the nature of the parsing error.
-        //           4. Append root to document.
-
-        dbgln("DOMParser::parse_from_string: Unimplemented parser for type: {}", Bindings::idl_enum_to_string(type));
-        TODO();
+
+        // 1. Create an XML parser parse, associated with document, and with XML scripting support disabled.
+        XML::Parser parser(string, { .resolve_external_resource = resolve_xml_resource });
+        XMLDocumentBuilder builder { document, false };
```
Comment (IdanHo) [inline Userland/Libraries/LibWeb/HTML/DOMParser.cpp:43]: maybe `XMLScriptingSupport::Disabled`?

File: Userland/Libraries/LibWeb/HTML/DOMParser.cpp [hunk 2]
```diff
@@ -36,16 +37,23 @@ NonnullRefPtr<DOM::Document> DOMParser::parse_from_string(String const& string,
...
+        // 2. Parse string using parser.
+        auto result = parser.parse_with_listener(builder);
+        // 3. If the previous step resulted in an XML well-formedness or XML namespace well-formedness error, then:
+        if (result.is_error() || builder.has_error()) {
+            // NOTE: The XML parsing can product nodes before it hits an error, just remove them.
```
Comment (IdanHo) [inline Userland/Libraries/LibWeb/HTML/DOMParser.cpp:48]: produce

Comment (alimpfard) [reply Userland/Libraries/LibWeb/HTML/DOMParser.cpp:48]: typos, typos

**Comment by awesomekling** [comment]
Nice work! :star:

**Pull request merged**